src/Array.mo

```diff
@@ -22,8 +22,9 @@ module {
     };
     return true;
   };
+
   /// Append the values of two input arrays
-  /// @deprecated Array.append has critical performance flaws; use a Buffer, and Buffer.append, instead.
+  /// @deprecated `Array.append` copies its arguments and has linear complexity; when used in a loop, consider using a `Buffer`, and `Buffer.append`, instead.
```

Comment:
Then this is not really "deprecated". It's more like a warning. A similar case is `Option.unwrap`, there are legitimate cases for using `unwrap`, so we need a warning instead of deprecated.

Reply:
Well the deprecation is actually reported as a warning too ;-> Are you suggesting we add another form of doc comment? We could do that, but not in this PR.

Reply:
Maybe @deprecated @expensive @trapping... or even @<whatever>